storage/rocksdb/ha_rocksdb.cc

```diff
@@ -6003,37 +6082,72 @@ int ha_rocksdb::read_key_exact(const Rdb_key_def &kd,
     lookup_tuple may be a prefix of the index.
   */
   rocksdb_smart_seek(kd.m_is_reverse_cf, iter, key_slice);
-  if (!iter->Valid() || !kd.value_matches_prefix(iter->key(), key_slice)) {
+
+  for (;;) {
```

Comment:
I think this loop can be rewritten more efficiently. Look at the below examples...